### PR TITLE
Add consent tracking model

### PR DESCRIPTION
**Pull request by agnessnowplow**
Add docs for the upcoming Consent Tracking custom module that will be added to the snowplow-web package.

### CODE REVIEW

File: docs/modeling-your-data/modeling-your-data-with-dbt/dbt-models/dbt-web-data-model/index.md [hunk 1]
```diff
@@ -24,3 +24,67 @@ This model consists of a series of modules, each producing a table which serves
...
+
+  - `snowplow_web_consent_users`: Incremental table of user consent tracking stats
+
+  - `snowplow_web_consent_totals`: Summary of the latest consent status as per consent version
+
+  - `snowplow_web_consent_scope_status`: Aggregate of current number of users consented to each consent scope
+
+  - `snowplow_web_cmp_stats`: Used for modeling cmp_visible events and related metrics
+
+  - `snowplow_web_consent_versions`: Incremental table used to keep track of each consent version and its validity
```
Comment (rlh1994) [inline docs/modeling-your-data/modeling-your-data-with-dbt/dbt-models/dbt-web-data-model/index.md:55]:
```suggestion
This custom module consists of a series of dbt models which produce the following aggregated models from the raw consent tracking events:

  - `snowplow_web_consent_log`: Snowplow incremental table showing the audit trail of consent and Consent Management Platform (cmp) events

  - `snowplow_web_consent_users`: Incremental table of user consent tracking stats

  - `snowplow_web_consent_totals`: Summary of the latest consent status, per consent version

  - `snowplow_web_consent_scope_status`: Aggregate of current number of users consented to each consent scope

  - `snowplow_web_cmp_stats`: Used for modeling cmp_visible events and related metrics

  - `snowplow_web_consent_versions`: Incremental table used to keep track of each consent version and its validity
```

File: docs/modeling-your-data/modeling-your-data-with-dbt/dbt-models/dbt-web-data-model/index.md [hunk 2]
```diff
@@ -24,3 +24,67 @@ This model consists of a series of modules, each producing a table which serves
...
+models:
+  snowplow_web:
+    optional_modules:
+      consent:
+        enabled: true
+        bigquery:
+          enabled: "{{ target.type == 'bigquery' | as_bool() }}"
+        snowflake:
+          enabled: "{{ target.type == 'snowflake' | as_bool() }}"
+```
```
Comment (rlh1994) [inline docs/modeling-your-data/modeling-your-data-with-dbt/dbt-models/dbt-web-data-model/index.md:78]: 💡 Can we maybe call out somewhere this is currently only available for bigquery and snowflake?

File: docs/modeling-your-data/modeling-your-data-with-dbt/dbt-models/dbt-web-data-model/index.md [hunk 3]
```diff
@@ -24,3 +24,67 @@ This model consists of a series of modules, each producing a table which serves
...
+
+####  Run the module
+If it is the first time you would like to process the snowplow_web package then you can run the package the recommended way either through your CLI or from within dbt Cloud with the following command:
+
+```
+dbt run --selector snowplow_web
+```
+
+However, assuming that you already have a working snowplow_web package and you would like to enable consent tracking and modeling at a later stage, you do not need to rebuild the whole model from scratch. Please refer to the [Custom module](/docs/modeling-your-data/modeling-your-data-with-dbt/dbt-custom-models/index.md) section for a detailed guide on how to achieve this the most efficient way.
```
Comment (rlh1994) [inline docs/modeling-your-data/modeling-your-data-with-dbt/dbt-models/dbt-web-data-model/index.md:87]:
```suggestion
####  Run the module
If you have previously run the web model without this optional module enabled, you can simply enable the module and run `dbt run --selector snowplow_web` as many times as needed for this module to catch up with your other data. If you only wish to process this from a specific date, be sure to change your `snowplow__start_date`, or refer to the [Custom module](/docs/modeling-your-data/modeling-your-data-with-dbt/dbt-custom-models/index.md) section for a detailed guide on how to achieve this the most efficient way.

If you haven't run the web package before, then you can run it using `dbt run --selector snowplow_web` either through your CLI, within dbt Cloud, or for Enterprise customers you can use the BDP console. In this situation, all models will start in-sync as no events have been processed.

```

File: docs/modeling-your-data/modeling-your-data-with-dbt/dbt-models/dbt-web-data-model/index.md [hunk 4]
```diff
@@ -24,3 +24,67 @@ This model consists of a series of modules, each producing a table which serves
...
+
+#### Good-to-know
+Whenever a new consent version is added to be tracked, the model expects an `allow_all` event in order to attribute the events to the full list of latest consent scopes. It is advisable to send a test event of that kind straight after deployment so that the model can process the data accurately.
```
Comment (rlh1994) [inline docs/modeling-your-data/modeling-your-data-with-dbt/dbt-models/dbt-web-data-model/index.md:90]: 💡 Maybe call this out a little earlier along with the pre-req of loader 4.0.0?

**Review by rlh1994 — CHANGES_REQUESTED** [review]
Just a load of random nitpick things, feel free to accept or ignore them depending on whether you think they are easier to read or just make things more complex.

**Comment by agnessnowplow** [comment]
Just noticed the Enhanced Consent docs are not yet in place so I removed the link (it was pointing at the old Consent plugin) and we can always add it back once it becomes available.

**Review by rlh1994 — APPROVED** [review]
Other than that one typo this looks good to me! Just gotta wait for #204 to go in first.

**Review by rlh1994 — APPROVED** [review]

**Pull request merged**